### DISCUSSION TITLE
Alternative download locations for cask

### DISCUSSION THREAD

**Discussion by fwenzel** [Homebrew/discussions#5576] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.18
ORIGIN: https://github.com/Homebrew/brew
HEAD: abc0584215659c5a8716d0798ea2801388d01368
Last commit: 2 days ago
Core tap JSON: 28 Aug 17:25 UTC
Core cask tap JSON: 28 Aug 17:25 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
﻿﻿﻿﻿Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
  youtube-dl

Warning: You have the following deprecated, official taps tapped:
  Homebrew/homebrew-cask-fonts
Untap them with `brew untap`.

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  certifi
```

### Description of issue

calibre (the ebook app) downloads are:

- notoriously slow
- large enough for that to matter

on the [official download page](https://calibre-ebook.com/download_osx), there is a main download location (the slow one) and always an alternative (binary release on github, which just maps to github's object storage (on azure, I figure) and is therefore very fast.

Is there a way to replace this location in the cask formula, or add two locations and make the alternative location the main one? calibre updates frequently fail for me because they time out etc, when they could be done in 2 seconds.

Thanks!

### Relevant casks

calibre

**Comment by p-linnane** [comment]
Thanks for the report. I've opened https://github.com/Homebrew/homebrew-cask/pull/183664 to use GitHub for new builds moving forward.

**Reply by p-linnane** [reply]
Apparently this is an issue that goes back over a decade and the maintainer of Calibre has not provided a good solution.
